ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -124,6 +128,108 @@ prop_Mempool_InvalidTxsNeverAdded setup =
         , txInMempool `notElem` txsInMempoolBefore
         ]
 
+-- | When the mempool is at capacity, test that 'addTxs' blocks when
+-- attempting to add one more transaction and that it unblocks when there is
+-- adequate space. Adequate space is made by adding all of the existing
+-- transactions to the ledger and removing them from the mempool.
+prop_Mempool_Capacity :: Property
```

Comment:
I'd like to expand on this test such that it works based on a randomly generated `MempoolCapacity` and I can also think of a few other "nice to have" `MempoolCapacity` tests. However, this might not be necessary if we'd like to merge this sooner rather than later.